recipes/opencascade/all/conanfile.py

```diff
@@ -135,9 +135,16 @@ def source(self):
     def generate(self):
         tc = CMakeToolchain(self)
 
-        # Inject C++ standard from profile since we have removed hardcoded C++ standard from upstream build files
-        if not valid_min_cppstd(self, self._min_cppstd):
-            tc.variables["CMAKE_CXX_STANDARD"] = self._min_cppstd
+        if Version(self.version) < "7.8.0":
+            # Inject C++ standard from profile since we have removed hardcoded C++ standard from upstream build files
+            if not valid_min_cppstd(self, self._min_cppstd):
+                tc.variables["CMAKE_CXX_STANDARD"] = self._min_cppstd
+        else:
+            if not valid_min_cppstd(self, self._min_cppstd):
+                tc.cache_variables["BUILD_CPP_STANDARD"] = self._min_cppstd
+            else:
+                cppstd = str(self.settings.compiler.cppstd).replace("gnu", "").upper()
+                tc.cache_variables["BUILD_CPP_STANDARD"] = f"C++{cppstd}"
```

Comment:
Use my version from https://github.com/conan-io/conan-center-index/pull/26087/files#diff-0af486202ce46be559999052379dde77ccc77a1278ddd19ede122ba8f4ead052R138

The `CMAKE_CXX_STANDARD` assignment was done for Conan 1 compatibility, and the cppstd is always defined in Conan 2
```suggestion
        if Version(self.version) >= "7.8.0":
            cppstd = str(self.settings.compiler.cppstd).replace("gnu", "").upper()
            tc.cache_variables["BUILD_CPP_STANDARD"] = f"C++{cppstd}"
```